deepspeed/utils/groups.py

```diff
@@ -481,7 +493,12 @@ def _get_sequence_parallel_group():
     global mpu
     if mpu is not None and hasattr(mpu, 'get_sequence_parallel_group'):
         return mpu.get_sequence_parallel_group()
-    return None
+    elif mesh_device is not None:
+        return mesh_device.get_group(mesh_dim="sequence_parallel")
+    else:
+        KeyError("No sequence parallel group found")
```

Comment:
Failing fast is preferred, so raising an exception is the right thing to do here (asking for the SP group when SP is not enabled should not happen).